react/react_forms/README.md

```diff
@@ -1,387 +1,477 @@
 
-## Framing & Background
+The answer is: Form values are kept track of by the document, and that's how they were accessible via the DOM. Read [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement) if you want more informatiion about what happens under the hood, but the jist is: When it comes to forms, the document has its own way to hold "state." Not anything close to a React state, but it keeps track of itself.
 
-Working with forms in react requires a little bit of extra effort. Because input fields have their own state, we have to decide how to handle that state in react. There are two main ways we can do this:
+Because the document keeps track of forms itself, we have to choose which state we want to use to keep track of our forms. Do we use the document's state, or React's? Our choice is between:
 
 **Controlled Components**
 
-This means _managing the form state in a react component_. In order to do this we have update the form state on change, and tie the value of the form field to the state itself, so that it updates visually when it changes.
+This means _managing the form's state in a React component's state rather than using the input fields' values_. In order to do this, we have to do two things: First, we have to update the component's state when the form changes. Second, we have to make sure that the input's field reflects React's state to the user and not the document's. This allows us to keep everything in React, and it allows us to more easily handle input errors. For example, if a user types an invalid email in a text input, we can flag that their email is incorrect as soon as they finish typing.
 
 **Uncontrolled Components**
 
-This means _not managing the form state_ in a component. Not having react hold the state makes it simpler to implement, but we need a way to retrieve the values from the form. Usually we do this when the form submits, or we click a button.
+This means _not managing the form's state_ using our component's state, but rather, using the form's own state. Not having React hold the state makes it simpler to implement, but we need a way to retrieve the values from the form. Usually, we do this when the form submits. This means that it's trickier to handle malformed user inputs immediately when they're made. Instead, we have to wait to look at their inputs when they submit the form, analyze whether they're wrong, and display an error message.
+
+# Code Examples
+
+Copy the contents of this folder to a local folder, open it in your code editor, and run `npm install`:
 
-## Code Example
+[React forms exercise](https://github.com/joinpursuit/Pursuit-Core-Web/tree/master/react/react_forms/lab)
```

Comment:
I got a 404 page not found on this link ?

Reply:
Good catch - @crymall can this link point to the external repo for this code example?